Release script changes & changelog 0.14.0

ch1bo opened the pull request:
* Fixed and refined `release.sh` script to work for 0.14.0

* Also cleaned up the changelog and added links to issues

---

<!-- Consider each and tick it off one way or the other -->
* [x] CHANGELOG updated or not needed
* [x] Documentation updated or not needed
* [x] Haddocks updated or not needed
* [x] No new TODOs introduced or explained herafter

ch1bo committed:
Fix sed expression on tutorial

ch1bo committed:
Allow release.sh to continue with unstaged changes

ch1bo committed:
Do not produce .bak files when running release.sh

ch1bo committed:
Check we are running release.sh on master

ch1bo committed:
Make checkout of release more robust

ch1bo committed:
Clean up changelog

v0d1ch reviewed (approved):
I tested the release script on my machine and it works :+1:

ch1bo committed:
Change tutorial to replace version more easily

ch1bo committed:
Update mithril release in tutorial

ch1bo committed:
Use hydra-node gen-hydra-key in tutorial

ch1bo committed:
Introduce a network.json file

This file is expected to hold transaction ids for released Hydra
versions. As such we can use it to make the tutorial simpler to be kept
up-to-date.

ch1bo committed:
Small typo in tutorial
